news-explorer-frontend/src/components/App/App.js

```diff
@@ -0,0 +1,394 @@
+  const [savedArticles, setSavedArticles] = useState([]);
+  const [isInfoTooltipOpen, setIsInfoTooltipOpen] = useState(false);
+  const [currentUser, setCurrentUser] = useState({
+    name: '',
+    id: '',
+  });
+  const [isSignInPopupOpen, setIsSignInPopupOpen] = useState(false);
+  const [isRegisterPopupOpen, setIsRegisterPopupOpen] = useState(false);
+  const [feedbackPopupOpenClass, setFeedbackPopupOpenClass] = useState('info-popup_open');
+  const [linkColorClass, setLinkColorClass] = useState('');
+  const [navBackgroundTypeClass, setNavBackgroundTypeClass] = useState('');
+  const [currentTypeClass, setCurrentTypeClass] = useState('');
+  const [mobileNavTypeClass, setMobileNavTypeClass] = useState('');
+  const [mobileLinkTypeClass, setMobileLinkTypeClass] = useState('');
+  const [navMobileBtnClass, setNavMobileBtnClass] = useState('');
+  const [headerDarkOverlay, setHeaderDarkOverlay] = useState('');
+  const [showPreloaderClass, setShowPreloaderClass] = useState('');
+  const navMenuOpenClass = isMobileNavOpen ? 'mobile-navigation_open' : '';
+  const location = useLocation();
+  const [token, setToken] = useState(localStorage.getItem('token'));
+  const [keyword, setKeyword] = useState('');
+  const [keywordsList, setKeywordsList] = useState({});
+  const [arrayOfKeywords, setArrayOfKeywords] = useState([]);
+  const navigate = useNavigate();
+  const path = location.pathname;
+
+  const orderKeywordsArray = () => {
+    const highestToLowest = Object.entries(keywordsList).sort((a, b) => b[1] - a[1]);
+    setArrayOfKeywords(highestToLowest);
+  };
+
+  const handleKeywordsList = () => {
+    let keywords = {};
+    savedArticles.forEach((article) => {
+      const newKeyword = {};
+      if (!(article.keyword in keywords)) {
+        newKeyword[article.keyword] = 1;
+        keywords = { ...keywords, ...newKeyword };
+      }
+      else {
+        keywords[article.keyword]++;
+      }
+    });
+    setKeywordsList(keywords);
+  };
+
+  useEffect(() => {
+    handleKeywordsList(savedArticles);
+  }, [savedArticles]);
+
+  useEffect(() => {
+    orderKeywordsArray();
+  }, [keywordsList]);
+
+  // after authorization get the content of the page
+  const tokenCheck = (jwt) => {
+    // check that the user has a valid token
+    if (jwt) {
+      getContent(jwt) // verifies the token
+        .then((res) => {
+          if (res) {
+            setCurrentUser({
+              name: res[0].name,
+              id: res[0]._id,
+            });
+            setLoggedIn(true);
+
+            navigate('/saved-news');
+          }
+        })
+        .catch((err) => console.log(err));
+    }
+  }; // Now, if a user logs in and leaves the page,
+  // when they return, everything will appear right where they left it
+
+  useEffect(() => {
+    api._headers = { authorization: `Bearer ${token}`, 'Content-Type': 'application/json' };
```

Comment:
`api._headers = `

Please, delete the `low dash` from methods if you want to use them `outside` a class because the `low dash` shows a `private` method which is used only inside the class.


or you can make another method `updateHeaders` in `api` and do it like this


```js

api.updateHeaders(token)

```

and everything should be set inside the class